x/claim/keeper/invariants.go

```diff
@@ -44,3 +60,43 @@ func AirdropSupplyInvariant(k Keeper) sdk.Invariant {
 		return "", false
 	}
 }
+
+// ClaimRecordMissionInvariant invariant checks that claim record completed missions exist
+func ClaimRecordMissionInvariant(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		missions := k.GetAllMission(ctx)
+		claimRecords := k.GetAllClaimRecord(ctx)
+
+		missionMap := make(map[uint64]struct{})
+		for _, mission := range missions {
+			missionMap[mission.MissionID] = struct{}{}
+		}
+		for _, claimRecord := range claimRecords {
+			for _, mission := range claimRecord.CompletedMissions {
+				if _, ok := missionMap[mission]; !ok {
+					return fmt.Sprintf("mission %d not exist", mission), true
+				}
+			}
+		}
+
+		return "", false
+	}
+}
+
+// InitialClaimMissionInvariant invariant checks the initial claim mission exist
+func InitialClaimMissionInvariant(k Keeper) sdk.Invariant {
```

Comment:
I actually don't know if we need this one.
This condition is checked in genesis validation and there is not state transitions in the chain that change the missions in the store.
To me, invariants are a protection to ensure the possible state transitions of the chain don't create an inconsistent state in the chain.
For example, `ClaimRecordMissionInvariant` makes total sense since claim records modified and completed missions are added dynamically through the on-chain transitions.

Just raising this to have some of your opinion. I don't think having an additional invariant influences the performances or the cost in `crisis` module since checking on-chain an invariant is done for an individual invariant